tests/Behat/FieldsContext.php

```diff
@@ -73,31 +65,83 @@ public function assertSelectOptions(string $select, TableNode $table): void {
+    $option = $this->unescapeStepArgument($option);
+    $position = $this->convertOrdinalToNumber($position) - 1;
+
+    $this->getNthField($field, $position)->selectOption($option);
+  }
+
+  /**
+   * Asserts that a field of a multi-value field item is marked as required.
+   *
+   * @param string $multi_value_field
+   *   The multiple cardinality field name.
+   * @param string $field
+   *   The name of the item field to assert.
+   * @param string $item
+   *   The ordinal number of the item. Defaults to the first item.
    *
    * @throws \Exception
-   *   Thrown when the specified occurrence of the list or the list itself is
-   *   not found.
+   *   Thrown when the element is not found.
+   * @throws \PHPUnit\Framework\ExpectationFailedException
+   *   Thrown when the element is not marked as required.
    *
-   * @Then I select :select from the :position :option
+   * @Then the :field field in the :item item of the :multi_value_field field should be marked as required
+   * @Then the :field field in the :multi_value_field field item should be marked as required
    */
-  public function selectNthOption(string $select, string $position, string $option): void {
-    $field = $this->unescapeStepArgument($option);
-    $select = $this->unescapeStepArgument($select);
-    $position = $this->convertOrdinalToNumber($position) - 1;
+  public function assertMultipleCardinalityFieldItemFieldMarkedAsRequired(string $multi_value_field, string $field, string $item = '1st'): void {
+    $multi_value_field = $this->unescapeStepArgument($multi_value_field);
+    $field = $this->unescapeStepArgument($field);
+    $item = $this->convertOrdinalToNumber($item) - 1;
+    $field_table = $this->getMultipleCardinalityFieldTable($multi_value_field);
+    $row = $this->findTableRow($field_table, $item);
+
+    $element_node = $row->findField($field);
+    if (!$element_node) {
+      throw new \Exception(sprintf('Cannot find element "%s" inside the %s item of the field table "%s".', $field, $item, $multi_value_field));
+    }
+
+    if (!$element_node->hasAttribute('required')) {
+      throw new ExpectationFailedException(sprintf('The element "%s" is not marked as required.', $field));
+    }
+  }
 
-    // Find all the fields with the specified name.
-    $fields = $this->getSession()->getPage()->findAll('named', ['field', $field]);
-    if (!$fields || !isset($fields[$position])) {
-      throw new \Exception(sprintf('Could not find field "%s" in position "%s".', $field, $position));
+  /**
+   * Asserts that a field of a multi-value field item is not marked as required.
+   *
+   * @param string $multi_value_field
+   *   The multiple cardinality field name.
+   * @param string $field
+   *   The name of the item field to assert.
+   * @param string $item
+   *   The ordinal number of the item. Defaults to the first item.
+   *
+   * @throws \PHPUnit\Framework\ExpectationFailedException
+   *   Thrown when the field is not marked as required.
```

Comment:
Thrown when the field is marked as required.

Reply:
Yep please go ahead and commit the fix :)